File: 狄泰软件学院/2 - C++深度解析教程/第17课 - 对象的构造（上）/class/code/17-1/17-1.cpp
#include <stdio.h>

class Test
{
private:
    int i;
    int j;
public:
    int getI() { return i; }
    int getJ() { return j; }
};

Test gt;

int main()
{
    // For global one, its members will be initialized
    // to 0 automatically
    printf("gt.i = %d\n", gt.getI());
    printf("gt.j = %d\n", gt.getJ());

    Test t1;

    // For normal local variable/class, they will be
    // random numbers if they are not initialized
    // accordingly
    printf("t1.i = %d\n", t1.getI());
    printf("t1.j = %d\n", t1.getJ());

    Test* pt = new Test;

    // In Windows, for members in heap memory, they will 
    // be random numbers if they are not initialized accordingly
    // In Linux, below should be random numbers, but somehow 
    // they are always 0
    printf("pt->i = %d\n", pt->getI());
    printf("pt->j = %d\n", pt->getJ());

    delete pt;

    return 0;
}

File: 狄泰软件学院/1 - C语言进阶剖析教程/第1期 - 初出茅庐（免费）/第17课 - ++和--操作符分析/class/code/17-1.java

class Test {
    public static void main(String[] args) {	
        int i = 0;
        int r = 0;

        r = (i++) + (i++) + (i++);

        System.out.println("i = " + i);
        System.out.println("r = " + r);

        r = (++i) + (++i) + (++i);

        System.out.println("i = " + i);
        System.out.println("r = " + r);
    }
}